### DISCUSSION TITLE
Is it possible to transform useFieldArray data to key value

### DISCUSSION THREAD

**Discussion by Abish-mako** [react-hook-form/react-hook-form#7837] [Q&A]
Can I transform useFieldArray data to key-value pairs?

useFieldArray array returns like this array of object

`
[
 {
     key: "valueinkey",
     value: "valueinvalue"
 }
]
`

Is it possible to transform something like this

`
[
{
     "valueinkey": "valueinvalue"
 }
]
`

[CSB](https://codesandbox.io/s/optimistic-joliot-qgloyr?file=/src/App.js)

**Comment by bluebill1049** [comment]
I think you will have to write a data transform at the submit function.
